mlflow/metrics/genai/model_utils.py

```diff
@@ -54,13 +54,23 @@ def _call_openai_api(openai_uri, payload):
             error_code=INVALID_PARAMETER_VALUE,
         )
 
+    config = {"openai_api_key": os.environ["OPENAI_API_KEY"]}
+    if "OPENAI_API_BASE" in os.environ:
+        config["openai_api_base"] = os.environ["OPENAI_API_BASE"]
+    if "OPENAI_API_TYPE" in os.environ:
+        config["openai_api_type"] = os.environ["OPENAI_API_TYPE"]
+    if "OPENAI_API_VERSION" in os.environ:
+        config["openai_api_version"] = os.environ["OPENAI_API_VERSION"]
+    if "OPENAI_DEPLOYMENT_NAME" in os.environ:
+        config["openai_deployment_name"] = os.environ["OPENAI_DEPLOYMENT_NAME"]
```

Comment:
what happens if the  user only provides `OPENAI_API_KEY` and `OPENAI_DEPLOYMENT_NAME` env vars but not the others? 

if that doesn't work, can we add some error handling if only certain env vars are set?

Reply:
We already throw this error:
```
2023/10/18 00:04:39 INFO mlflow.metrics.genai.genai_metric: Failed to score model on payload. Error: MlflowException("OpenAI route configuration must specify 'openai_api_base', 'openai_deployment_name', and 'openai_api_version' if 'openai_api_type' is 'azure' or 'azuread'.")
```

Comes from here: https://github.com/mlflow/mlflow/blob/a4db4ee826765f0365944f83124af8840c72f4d8/mlflow/gateway/config.py#L138